src/parser.ts

```diff
@@ -68,44 +68,45 @@ function parseNonLiteral(
 ) {
 
   log(whiteBright.bgBlue('parser'), schema, '<-' + typeOfSchema(schema), processed.has(schema) ? '(FROM CACHE)' : '')
+  const customName = () => { return ( options.customName ? options.customName({schema, keyName, definitionKeyName: keyNameFromDefinition, usedNames}) : undefined ) }
```

Comment:
I think for consistency it would be nice to pull up the `customName() || standaloneName()` parts into a `name` here as well, in the same way it is done later?

Reply:
customName and standaloneName potentially change state (i.e. it marks a name as being used) and the result would not be used in cases where `newInterface` is called. I guess though I could change `newInterface` signature and just pass the generated name through, but there is still potential for problems. The other option is to make a function wrapper `let name = () => { return customName() || standaloneName(schema, keyNameFromDefinition, usedNames)! }` - will look into this maybe a bit later.

Reply:
@ankon I have simplified it slightly - have a look and tell me what you think.